From d2943c4ee1c52cb2c9802be194ca90442e4fe34f Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.org>
Date: Tue, 16 Feb 2016 08:47:56 +0000
Subject: [PATCH 153/170] Revert "Add blk_pos parameter to mmc multi_io_quirk
 callback"

This reverts commit aab95f9b10e8c3d32de2bf163b86f220c88214fe.

The bcm2835-sdhost driver no longer needs this patch.
---
 drivers/mmc/card/block.c          | 1 -
 drivers/mmc/host/omap_hsmmc.c     | 4 +---
 drivers/mmc/host/sh_mobile_sdhi.c | 4 +---
 drivers/mmc/host/tmio_mmc_pio.c   | 4 +---
 include/linux/mmc/host.h          | 4 +---
 5 files changed, 4 insertions(+), 13 deletions(-)

--- a/drivers/mmc/card/block.c
+++ b/drivers/mmc/card/block.c
@@ -1510,7 +1510,6 @@ static void mmc_blk_rw_rq_prep(struct mm
 			brq->data.blocks = card->host->ops->multi_io_quirk(card,
 						(rq_data_dir(req) == READ) ?
 						MMC_DATA_READ : MMC_DATA_WRITE,
-						blk_rq_pos(req),
 						brq->data.blocks);
 	}
 
--- a/drivers/mmc/host/omap_hsmmc.c
+++ b/drivers/mmc/host/omap_hsmmc.c
@@ -1832,9 +1832,7 @@ static void omap_hsmmc_conf_bus_power(st
 }
 
 static int omap_hsmmc_multi_io_quirk(struct mmc_card *card,
-				     unsigned int direction,
-				     u32 blk_pos,
-				     int blk_size)
+				     unsigned int direction, int blk_size)
 {
 	/* This controller can't do multiblock reads due to hw bugs */
 	if (direction == MMC_DATA_READ)
--- a/drivers/mmc/host/sh_mobile_sdhi.c
+++ b/drivers/mmc/host/sh_mobile_sdhi.c
@@ -170,9 +170,7 @@ static int sh_mobile_sdhi_write16_hook(s
 }
 
 static int sh_mobile_sdhi_multi_io_quirk(struct mmc_card *card,
-					 unsigned int direction,
-					 u32 blk_pos,
-					 int blk_size)
+					 unsigned int direction, int blk_size)
 {
 	/*
 	 * In Renesas controllers, when performing a
--- a/drivers/mmc/host/tmio_mmc_pio.c
+++ b/drivers/mmc/host/tmio_mmc_pio.c
@@ -1003,9 +1003,7 @@ static int tmio_mmc_get_ro(struct mmc_ho
 }
 
 static int tmio_multi_io_quirk(struct mmc_card *card,
-			       unsigned int direction,
-			       u32 blk_pos,
-			       int blk_size)
+			       unsigned int direction, int blk_size)
 {
 	struct tmio_mmc_host *host = mmc_priv(card->host);
 
--- a/include/linux/mmc/host.h
+++ b/include/linux/mmc/host.h
@@ -143,9 +143,7 @@ struct mmc_host_ops {
 	 * I/O. Returns the number of supported blocks for the request.
 	 */
 	int	(*multi_io_quirk)(struct mmc_card *card,
-				  unsigned int direction,
-				  u32 blk_pos,
-				  int blk_size);
+				  unsigned int direction, int blk_size);
 };
 
 struct mmc_card;
